airflow/example_dags/example_trigger_target_dag.py

```diff
@@ -24,11 +24,12 @@
 from datetime import datetime
 
 from airflow import DAG
+from airflow.decorators import task
 from airflow.operators.bash import BashOperator
-from airflow.operators.python import PythonOperator
 
 
-def run_this_func(dag_run):
+@task(task_id="run_this")
+def run_this_func(dag_run=None):
```

Comment:
This would fail if there's no dag_run because of the print statement below

Reply:
Yes it sure would. Thanks for the catch!

Reply:
Actually this is supposed to be used in tandem with the `example_trigger_controller_dag` DAG which triggers this DAG based on the docstring:
```
"""
Example usage of the TriggerDagRunOperator. This example holds 2 DAGs:
1. 1st DAG (example_trigger_controller_dag) holds a TriggerDagRunOperator, which will trigger the 2nd DAG
2. 2nd DAG (example_trigger_target_dag) which will be triggered by the TriggerDagRunOperator in the 1st DAG
"""
```
Verified the "controller" DAG has the `TriggerDagRunOperator` sending a `conf` arg:
```python
trigger = TriggerDagRunOperator(
        task_id="test_trigger_dagrun",
        trigger_dag_id="example_trigger_target_dag",  # Ensure this equals the dag_id of the DAG to trigger
        conf={"message": "Hello World"},
    )
```

Do you think it's still worth adding a simple check on `dag_run.conf["message"]`?

Reply:
From a syntax standpoint, the dag_run should be a required value if we access its attribute as we did. I would suggest we change it but I don't have strong opinion on it

Reply:
I see, good point. I feel it's better to show that you don't need to explicitly pass in the `dag_run` object (any context object for that matter) when using the TaskFlow API (hoping it's another example that makes the TaskFlow API more approachable). I think updating the access method would be a better move.